extension/awsecshealthcheckextension/factory.go

```diff
@@ -0,0 +1,56 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsecshealthcheckextension
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/extension/extensionhelper"
+)
+
+const (
+	// The value of extension "type" in configuration.
+	typeStr = "aws_ecs_health_check"
+
+	defaultEndpoint = "127.0.0.1:13134"
+)
+
+// NewFactory creates a factory for HealthCheck extension.
+func NewFactory() component.ExtensionFactory {
+	return extensionhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		createExtension)
+}
+
+func createDefaultConfig() config.Extension {
+	return &Config{
+		ExtensionSettings: config.NewExtensionSettings(config.NewID(typeStr)),
+		TCPAddr: confignet.TCPAddr{
+			Endpoint: defaultEndpoint,
+		},
+		Interval:           "5m",
+		ExporterErrorLimit: 5,
+	}
+}
+
+func createExtension(_ context.Context, set component.ExtensionCreateSettings, cfg config.Extension) (component.Extension, error) {
+	config := cfg.(*Config)
+
+	return newServer(*config, set.Logger), nil
```

Comment:
```suggestion
	return newServer(*config, set), nil
```

I'd recommend taking the settings object directly, that gives more flexibility for the future if other settings are useful.  For instance, this will have access to a `TracerProvider` that may be of use since this extension starts an HTTP server.

Reply:
all extensions use the same newExtension/newServer function with return the set.logger instead of set, should we do this special?

Reply:
This is something I'm expecting to change in the near future.  See, for example, [the changes made here](https://github.com/open-telemetry/opentelemetry-collector/pull/3982/files#diff-00632bae2b9437133015b45877fc74e3006d5f26d895a4839b833c52e74fc0e6) to enable passing those settings to `obsreport`.  Making the change here now will help minimize the churn when we have to update contrib.